channeldb/payment_control_test.go

```diff
@@ -454,11 +472,37 @@ func checkPaymentAttemptInfo(bucket *bbolt.Bucket, a *PaymentAttemptInfo) error
 	if err != nil {
 		return err
 	}
+
+	for i := 0; i < len(a.Route.Hops); i++ {
+		for j := 0; j < len(a.Route.Hops[i].TLVRecords); j++ {
+			expectedRecord := a.Route.Hops[i].TLVRecords[j]
+			newRecord := a2.Route.Hops[i].TLVRecords[j]
+
+			err := assertHopRecordsEqual(expectedRecord, newRecord)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	recordCache := make(map[int][]tlv.Record)
+	for i := 0; i < len(a.Route.Hops); i++ {
+		recordCache[i] = a.Route.Hops[i].TLVRecords
+
+		a.Route.Hops[i].TLVRecords = nil
+
+		a2.Route.Hops[i].TLVRecords = nil
+	}
```

Comment:
Why is this needed? Could use a comment and/or extract to a function that does the caching, deepequal and restore.

Reply:
Added a comment. This case is slightly different from the other occurrence, so I don't think it needs a function quite yet.

Reply:
They may be a little different, but the goal in both cases is to compare two routes. I think that `assertRoutesEqual` is a useful function to have and also make the test slightly more readable.